nestkernel/simulation_manager.cpp

```diff
@@ -443,20 +507,39 @@ nest::SimulationManager::simulate( Time const& t )
       "is called repeatedly with simulation times that are not multiples of "
       "the minimal delay." );
 
-  resume_( num_active_nodes );
+  start_updating_();
+}
 
-  finalize_simulation_();
+void
+nest::SimulationManager::cleanup()
+{
```

Comment:
An earlier version had an enum that made a state machine checking that prepare, run and cleanup were called in the right order. I saw that PR #542 was playing with that, so I removed until merge. Apparently merged, so I should look again at exception throwing in terms of state.